src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RandomNumberGenerator.cs

```diff
@@ -24,20 +20,36 @@ public void Dispose()
             return;
         }
 
-        protected virtual void Dispose(bool disposing)
-        {
-            return;
-        }
+        protected virtual void Dispose(bool disposing) { }
 
         public abstract void GetBytes(byte[] data);
 
-        internal void ValidateGetBytesArgs(byte[] data)
+        public virtual void GetBytes(byte[] data, int offset, int count)
         {
-            if (data == null)
+            VerifyGetBytes(data, offset, count);
+            if (count > 0)
             {
-                throw new ArgumentNullException(nameof(data));
+                // For compat we can't avoid an alloc here since we must call GetBytes(data).
+                // However RandomNumberGeneratorImplementation avoids extra allocs.
+                byte[] tempData = new byte[count];
+                GetBytes(tempData);
+                Array.Copy(tempData, 0, data, offset, count);
```

Comment:
I understand we can't avoid the allocation/copy in the general case.  But could we for the specific case where the offset/count are for the whole array?
```C#
VerifyGetBytes(data, offset, count);
if (count > 0)
{
    if (offset == 0 && count == data.Length)
    {
        GetBytes(data);
    }
    else
    {
        var tempData = new byte[count];
        GetBytes(tempData);
        Buffer.BlockCopy(tempData, 0, data, offset, count);
    }
}
```